kurtosis-devnet/tests/eigenda/failover_test.go

```diff
@@ -0,0 +1,409 @@
+package eigenda_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/Layr-Labs/eigenda-proxy/clients/memconfig_client"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/geth"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/kurtosis-tech/kurtosis/api/golang/core/lib/enclaves"
+	"github.com/kurtosis-tech/kurtosis/api/golang/engine/lib/kurtosis_context"
+	"github.com/stretchr/testify/require"
+)
+
+// All tests are run in the context of the eigenda-memstore-devnet enclave.
+// We assume that this enclave is already running.
+const enclaveName = "eigenda-memstore-devnet"
+
+// TestFailover tests the failover behavior of the batcher, in response to the proxy returning 503 errors.
+// See https://github.com/Layr-Labs/eigenda-proxy?tab=readme-ov-file#failover-signals for proxy behavior.
+// The proxy's memstore's failover behavior is toggled on and off by this test via a REST api.
+// We then check that the batcher correctly interprets the 503 signals and starts submitting batches to EthDA instead.
+// The test then toggles the failover back off and checks that the batcher starts submitting EigenDA batches again.
+// The batches inbox transactions are queried via geth's GraphQL API.
+// TODO: We will also need to test the failover behavior of the node, which currently doesn't finalize after failover (fixed in https://github.com/Layr-Labs/optimism/pull/23)
+func TestFailover(t *testing.T) {
```

Comment:
will we eventually wanna add tests ensuring failover to?
- keccak commitment mode
- 4844

Reply:
keccak yes probably, I'll add a comment. 4844 I don't know... it's hard to implement and not a priority for me.

Reply:
[bedc45f](https://github.com/Layr-Labs/optimism/pull/33/commits/bedc45f56bd36c8297815e34d9b969413997c2a0)